### PR TITLE
fix: subscription modifiers inherit plan defaults

### PR DESCRIPTION
**Pull request by brianium**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,19 @@
 {
-    "phpclassgen.vendor": "TeamGantt\\Dues"
+    "phpclassgen.vendor": "TeamGantt\\Dues",
+    "workbench.colorCustomizations": {
```
Comment (johncorrelli) [inline .vscode/settings.json]: Excuse me?

File: src/Model/Subscription.php [hunk 2]
```diff
@@ -286,8 +287,35 @@ public function setPlan(?Plan $plan): self
...
+    private function mergeCurrentModifiers(Modifiers $source, array $modifiers): Modifiers
+    {
+        $provided = $source->getAll();
+        $modifierDefaults = new Modifiers($this, $modifiers);
+
+        foreach ($provided as $modifier) {
+            $modifierDefaults->add($modifier);
+        }
+
+        return $modifierDefaults;
```
Comment (johncorrelli) [inline src/Model/Subscription.php:319]: Nice!

**Review by johncorrelli — CHANGES_REQUESTED** [review]
Just a weird git thing with `.vscode` dir that would be nice to tidy up, if you're up to it... but looks groovy. and nice work on the modifiers.

File: .vscode/settings.json (hunk 1 above)
Comment (brianium) [reply .vscode/settings.json]: I'm removing the vscode settings in a follow-up commit

Comment (brianium) [reply .vscode/settings.json]: live share patches the settings for live share colors and that could get super annoying to track 😱

**Review by johncorrelli — APPROVED** [review]

**Pull request merged**